apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -71,23 +71,79 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
   def get_branch_route_stops(route, direction_id, route_pattern_id) do
     route
     |> do_get_branch_route_stops(direction_id, route_pattern_id)
-    |> RouteStop.list_from_route_patterns(route, direction_id)
-    |> Enum.chunk_by(& &1.branch)
+    |> Enum.map(&RouteStop.list_from_route_pattern(&1, route, direction_id))
+    |> make_trunks_consistent(route)
     |> RouteStops.from_route_stop_groups()
   end
 
   @spec do_get_branch_route_stops(Route.t(), direction_id(), RoutePattern.id_t() | nil) :: [
           {RoutePattern.t(), [Stop.t()]}
         ]
   defp do_get_branch_route_stops(route, direction_id, route_pattern_id) do
-    route_patterns = get_line_route_patterns(route.id, direction_id, route_pattern_id)
-
-    route_patterns
+    route.id
+    |> get_line_route_patterns(direction_id, route_pattern_id)
     |> filtered_by_typicality(route, route_pattern_id)
     |> Enum.map(&stops_for_route_pattern/1)
     |> maybe_use_overarching_branch()
   end
 
+  @routes_with_trunk_discrepancies ~w(CR-Franklin CR-Providence)
```

Comment:
It would be nice if we could detect trunk discrepancies instead of hard-coding the routes, but I also like this because it ensures this special case code doesn't affect other routes.

Reply:
Yeah, in theory / eventually I definitely agree. But as you say, I was afraid of causing any unintended side effects on other routes. Maybe eventually we can build up enough test cases to be confident that we are able to detect this one form of discrepancy without any false positives.